refinery/core/api.py

```diff
@@ -344,7 +347,15 @@ def res_sharing(self, request, **kwargs):
                 is_read_only = can_read and not can_change
                 should_share = can_read or can_change
 
-                if should_share:
+                # datasets handled seperate due to object-level perm
+                if self.res_type._meta.verbose_name == 'dataset':
+                    is_read_meta_only = False
+                    if not should_share and i['read_meta']:
+                        is_read_meta_only = i['read_meta']
+                        should_share = is_read_meta_only
```

Comment:
I'm not following the logic here: if we get to this point, then `i['read_meta']` is true (or trueish?)... so `is_read_meta_only` and `should_share` will always be true... and we'll always take the `if should_share` just below?